tools/functional-tester/etcd-tester/lease_stresser.go

```diff
@@ -0,0 +1,278 @@
+// Copyright 2015 The etcd Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package main
+
+import (
+	"fmt"
+	"math/rand"
+	"sync"
+	"time"
+
+	"github.com/coreos/etcd/clientv3"
+	"github.com/coreos/etcd/etcdserver/api/v3rpc/rpctypes"
+	pb "github.com/coreos/etcd/etcdserver/etcdserverpb"
+	"golang.org/x/net/context"
+	"google.golang.org/grpc"
+)
+
+// time to live for lease
+const TTL = 15
+
+type LeaseStresser interface {
+	// Stress starts to stress lease on the etcd cluster
+	Stress() error
+	// Cancel cancels the lease stress test on the etcd cluster
+	Cancel()
+	// Report reports the success and failure of the lease stress test
+	Report() (success int, failure int)
+	// Setup initializes fields of leaseStresser struct
+	Setup() error
+}
```

Comment:
> I don't want to cancel() to be empty for lease stresser because i want it to be cancelled on cluster.terminate() is called

You could type-assert on `Stresser` interface and skip canceling lease `Stresser`.